tests/common/test_gtsfm_data.py

```diff
@@ -0,0 +1,149 @@
+        """Test for total number of images."""
+        self.assertEqual(EXAMPLE_DATA.number_images(), 3)
+
+    def test_number_tracks(self):
+        """Test for number of tracks."""
+        self.assertEqual(EXAMPLE_DATA.number_tracks(), 7)
+
+    def test_get_valid_camera_indices_on_consecutive_indices(self):
+        """Tests on getter for valid camera indices when input has consecutive indices."""
+        expected = [0, 1, 2]
+        self.assertListEqual(EXAMPLE_DATA.get_valid_camera_indices(), expected)
+
+    def test_get_valid_camera_indices_on_nonconsecutive_indices(self):
+        """Test on getter for valid cameras indices when input has non-consecutive indices."""
+        expected = [0, 2, 3]
+        self.assertListEqual(EXAMPLE_WITH_NON_CONSECUTIVE_CAMS.get_valid_camera_indices(), expected)
+
+    def test_get_camera_valid(self):
+        """Test for get_camera for a valid index."""
+        expected = PinholeCameraCal3Bundler(Pose3(), Cal3Bundler(fx=900, k1=0, k2=0, u0=100, v0=100))
+
+        cam_idx = 0
+        test_data = GtsfmData(1)
+        test_data.add_camera(expected, cam_idx)
+
+        computed = test_data.get_camera(cam_idx)
+        self.assertTrue(computed.equals(expected, EQUALITY_TOLERANCE))
+
+    def test_get_camera_invalid(self):
+        """Test for get_camera for an index where the camera does not exist."""
+
+        computed = EXAMPLE_DATA.get_camera(10)
+        self.assertIsNone(computed)
+
+    def test_get_track(self):
+        """Testing getter for track."""
+        expected_track = SfmTrack(np.array([6.41689062, 0.38897032, -23.58628273]))
+        expected_track.add_measurement(0, np.array([383.88000488, 15.2999897]))
+        expected_track.add_measurement(1, np.array([559.75, 106.15000153]))
+
+        computed = EXAMPLE_DATA.get_track(1)
+
+        # comparing just the point because track equality is failing
+        np.testing.assert_allclose(computed.point3(), expected_track.point3())
+
+    def test_add_track_with_valid_cameras(self):
+        """Testing track addition when all cameras in track are already present."""
+
+        gtsfm_data = copy.deepcopy(EXAMPLE_DATA)
+
+        # add a track on camera #0 and #1, which exists in the data
+        track_to_add = SfmTrack(np.array([0, -2.0, 5.0]))
+        track_to_add.add_measurement(0, np.array([20.0, 5.0]))
+        track_to_add.add_measurement(1, np.array([60.0, 50.0]))
+
+        self.assertTrue(gtsfm_data.add_track(track_to_add))
+
+    def test_add_track_with_nonexistant_cameras(self):
+        """Testing track addition where some cameras are not in tracks, resulting in failure."""
+        gtsfm_data = copy.deepcopy(EXAMPLE_DATA)
+
+        # add a track on camera #0 and #1, which exists in the data
+        track_to_add = SfmTrack(np.array([0, -2.0, 5.0]))
+        track_to_add.add_measurement(0, np.array([20.0, 5.0]))
+        track_to_add.add_measurement(3, np.array([60.0, 50.0]))  # this camera does not exist
```

Comment:
keyword arg would help, to explain what 3 represents

Reply:
is it better now?

Reply:
much better!